sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -71,7 +71,9 @@ class DataFrameReader private[sql](sqlContext: SQLContext) extends Logging {
    * @since 1.4.0
    */
   def option(key: String, value: String): DataFrameReader = {
-    this.extraOptions += (key -> value)
+    if (value != null) {
```

Comment:
BTW, without this null check, the test case I added will fail due to null pointer exception.